### DISCUSSION TITLE
Why is master red most of the time?

### DISCUSSION THREAD

**Discussion by macsmol** [zigtools/zls#495] [Q&A] [answered]
Hi everyone. Today I've decided to give Zig a go.. Soon I realized I'll need a some way to debug the zig code if I'll be writing anything moderately complicated. So that's how I came here... 
I learned that there's this LSProtocol and that there's IntelliJ plugin to support LSP clients like this one(I come from Java background). 
~But I guess I'm stuck with using plugin for VS Code since the master of ZLS is failing for most of the time.~
Edit: Oh wait! zls-vscode is a different thing - It's necessary to compile zls anyway.

Ok. So how do I compile zls? I have 9351ca722d8a0cac8a530db0bd78b1a5944a6aa commit checked out and I get the same errors as in the last master build: https://github.com/zigtools/zls/runs/6640490340?check_suite_focus=true

My zig compiler version is zig-windows-x86_64-0.10.0-dev.2431+0e6285c8f 

BTW. Why people keep merging PRs on top of red master? Specifically why they do this for weeks?
Sorry for whining in the first post.. but it looks like it looks..

**Comment by truemedian** [comment, marked as answer]
> Why people keep merging PRs on top of red master? Specifically why they do this for weeks?

PRs have continued to be merged because we *knew* exactly what was broken. It was a specific change in a few of the things that zls uses, and we were just waiting for a patch that solved the problem (without introducing more).

Now that #496 is merged, zls is back to working on the latest zig master. Unfortunately zig makes these changes quite often.
